contracts/zero-ex/contracts/src/transformers/FillQuoteTransformer.sol

```diff
@@ -388,31 +387,110 @@ contract FillQuoteTransformer is
         private
         returns (FillOrderResults memory results)
     {
-        // Track changes in the maker token balance.
-        uint256 initialMakerTokenBalance = makerToken.balanceOf(address(this));
-        try
-            exchange.fillOrder
-                {value: protocolFee}
-                (order, takerAssetFillAmount, signature)
-            returns (IExchange.FillResults memory fillResults)
-        {
-            // Update maker quantity based on changes in token balances.
-            results.makerTokenBoughtAmount = makerToken.balanceOf(address(this))
-                .safeSub(initialMakerTokenBalance);
-            // We can trust the other fill result quantities.
-            results.protocolFeePaid = fillResults.protocolFeePaid;
-            results.takerTokenSoldAmount = fillResults.takerAssetFilledAmount;
-            // If the taker fee is payable in the taker asset, include the
-            // taker fee in the total amount sold.
-            if (isTakerFeeInTakerToken) {
-                results.takerTokenSoldAmount =
-                    results.takerTokenSoldAmount.safeAdd(fillResults.takerFeePaid);
+        bytes4 makerAssetProxyId = order.makerAssetData.readBytes4(0);
```

Comment:
Might need to bump this up to `_sellToOrder` and `_buyFromOrder` to prevent adjustments in amounts if fees are set on an order. Otherwise the behaviour is pretty undefined.